### DISCUSSION TITLE
'SCRATCHING' sample on gold knob (sample playback option)

### DISCUSSION THREAD

**Discussion by worker-gray** [SynthstromAudible/DelugeFirmware#1184] [Ideas]
I would like the ability to assign a 'scratch' playback option to a sample wherein when that sample was selected via the audition pad, it could be 'scratched' like a record on a turntable by using a gold knob- or any knob as long as the audition pad was depressed at the time. Otherwise the knobs would revert to their normal functions.

**Comment by worker-gray** [comment]
alternately, a combination of audition (arm) and mute (cue) could be held while 'scratching' with the knob. This would have the advantage of  being able to 'stutter' the scratch by rapidly tapping the mute (crossfader) while you whip the knob back and forth.
Audio example:
https://www.youtube.com/watch?v=lDp6H7EPjBo

**Comment by worker-gray** [comment]
another possibility is support for external turntable controllers so people with a preferred device can hook it up and scratch away.

**Comment by sapphire-arches** [comment]
There was some discussion of using xwax for something like this (https://github.com/SynthstromAudible/DelugeFirmware/discussions/277). Using the gold knobs instead of an external controller sidesteps some of the challenges but not all. One thing I didn't think about in that thread is the challenges related to sample loading -- the existing logic assumes only forward/reverse linear playback. If the scratch effect is limited to only +/- a small number of clusters (a handful of seconds) this could work, at the cost of a bunch of voices due to memory pressure.

I think the gold knob based implementation is technically an interesting direction to at least prototype.

**Reply by worker-gray** [reply]
I was a teenager in the heyday of turntablism (mid 80's) and at most they would scratch a word or two. Sometimes a short sentence. Some folks might be tempted to push that maximum, but I think the 'arm/cue/twist' approach would tend to cut-down on too many samples being used at once, plus when the 'arm/cue' (audition/mute) buttons are released, the software could automatically de-allocate whatever memory was required for the given sample.

**Comment by worker-gray** [comment]
it would also be convenient if there were some way to alter the transport speed of the gold knob for longer samples-- so rather than having to make a full revolution to go from one end of the sample to the other, you could set the speed so that you could cover the range with 3/4 or half a twist.
I'm thinking the detents in the encoder might cause skips or jumps in playback, since they are the only way the device knows they're being turned at all, isn't it?
Forgive me if I'm speaking above my paygrade. I don't know anything about the architecture of the deluge or it's firmware characteristics except the little I've gleaned reading these threads. I'm shooting from the hip, blindfolded in a dark room.

**Comment by worker-gray** [comment]
Just checking if there have been any advancements on this front.

**Comment by seangoodvibes** [comment]
> Just checking if there have been any advancements on this front.

Not as part of the community firmware

There are alternative firmwares out there that may have what you're looking for

One example: https://github.com/owlet-labs/DelugeFirmware/blob/dev/README.md

**Reply by worker-gray** [reply]
Is the 'owlet' firmware built on the latest community firmware release?

**Reply by seangoodvibes** [reply]
> Is the 'owlet' firmware built on the latest community firmware release?
> 
> 

It's built off the nightly build so it should have everything we have but it's its own branch managed by someone else.
